### PR TITLE
fix(time): convert the millis() rollover sites #11291's CI guard cannot see

### PR DESCRIPTION
**Pull request by thebentern**
Follow-up to #11291.

That PR fixed every naive `millis()` comparison in `src/` and added the `millis-deadline-check` CI guard to stop new ones landing. The guard matches a literal `millis()` / `getMillis()` token sitting next to a comparison operator, which is exactly the shape of the sites it was written against — and `develop` is green on it today (two hits, both allowlisted).

But a site that stashes the clock in a local or a member first walks straight past it:

```cpp
auto now = millis();
if (backPacket->tx_after < now && ...)   // same bug, invisible to the guard
```

I swept for every identifier in `src/` assigned from an expression containing `millis()`/`getMillis()`, then looked for comparisons on those identifiers that don't subtract first. This is what came back.

## Rollover bugs

| Site | Symptom |
| --- | --- |
| `MeshPacketQueue::replaceLowerPriorityPacket()` | `backPacket->tx_after < now` plus an ordering of two absolute deadlines. Across the wrap a late packet reads as not-yet-due for the rest of the window, or every late packet reads as droppable at once. |
| `ExpressLRSFiveWay::runOnce()` | `now > alertingSinceMs + 2000` and `now < keyDownStart + 20000`. `update()` in the same file was already on `Throttle`; these two were missed. |
| `GPSUpdateScheduling::elapsedSearchMs()` | "Am I searching?" asked as `searchStartedMs > searchEndedMs` — an ordering of two raw stamps, which inverts in **both** directions across the wrap. Either `searchedTooLong()` aborts a search that isn't running, or it never aborts one that is. Self-heals at the next `informSearching()`. |
| `RadioLibInterface` `TRANSMIT_DELAY_COMPLETED` | Not a wrap bug — a width bug. `long delay_remaining = txp->tx_after - millis()` is a `uint32_t` subtraction: on 32-bit targets an already-due deadline lands negative and the packet transmits, but where `long` is 64-bit (portduino, native tests) it zero-extends to ~4.29e9 and the packet is rescheduled 49.7 days out. `notifyLater()` fires at *or after* the deadline, so being a millisecond past due is the ordinary path here, not an edge case. |

## Fail-safe, but the wrong storage

`MessageStore` / `MessageRenderer` stamp boot-relative messages as `millis() / 1000` and test them with `m.timestamp <= bootNow`. Across the wrap a pre-wrap stamp reads as newer than "now", so `upgradeBootRelativeTimestamps()` declines to heal it and the message shows `???` until it ages out. No wrong time is ever printed — the guard is what fails — but `Time::getUptimeSecs()` landed in #11291 for precisely this and doesn't wrap for 136 years.

## Competing idioms removed

- `MessageStore`'s `reachedMs(now, target)` — `(int32_t)(now - target) >= 0`, a local reimplementation of `Throttle::deadlinePassedAt()` in the signed-cast form #11291 replaced everywhere else.
- `WebServer::getAdaptiveInterval()`'s hand-rolled wrap branch: `(UINT32_MAX - lastActivityTime) + currentTime + 1` computes the same number unsigned subtraction already gives. Not a bug — but eight lines that *look* like a site which has considered the wrap and settled it, which makes it a bad example to copy.

Every converted site now takes its stamp from `Time::getMillis()`, so writes and reads share the clock `Throttle` compares against and each one is reachable from a native test.

## Deliberately left alone

- **`GPSUpdateScheduling::msUntilNextSearch()`**'s `int32_t remainingMs = compensatedStart - now`. It needs the remaining magnitude, not a boolean — the same reason `MotionSensor`'s calibration countdown kept a signed delta in #11291.
- **`Screen.cpp`**'s `uint64_t startUpdate = millis()` in the two `updateUiFrame` loops. `startUpdate` is reassigned every iteration, so a wrap costs one extra frame and self-corrects. The `uint64_t` buys nothing, but that's cosmetic and the file belongs to the UI work.
- **Widening the CI guard** to catch the local-variable form. I prototyped it; separating `now - last > interval` (correct) from `now > last + interval` (broken) needs more than a line-local regex, and every version cheap enough for CI had a false-positive rate that would train people to reach for the allowlist. The sweep that found these is a scripted audit, which is the right tool for a periodic pass — not a per-PR gate.

## Tests

`test/test_gps_update_scheduling/` gains seven cases driving `Time::setTestMillis()` across the wrap: the idle / searching / ended states, elapsed exactness over the boundary, both inversion directions, and `reset()`. The two wrap cases fail against the old predicate.

`test/test_meshpacket_queue/` is new, covering the late-packet eviction branch: a future-dated arrival must not displace an overdue packet, both directions of the both-expired ordering, the undelayed arrival, and all of it again with the deadlines and `now` on opposite sides of the wrap. `maxLen` is 1 so the suite reaches the branch without dragging in `CompareMeshPacketFunc` and a `NodeDB`.

The remaining sites resist a native test for the usual reasons — `WebServer` and `MessageStore` are behind `HAS_SCREEN` / ESP32 WiFi, and `RadioLibInterface` needs the radio. The comparisons themselves are `Throttle`'s now, and `test/test_throttle/` already pins those across the wrap.

## Verification

- `millis-deadline-check` guard: still 2 hits, both allowlisted.
- Native suite green in CI, 54 suites, including both new/extended ones (`test_meshpacket_queue`, `test_gps_update_scheduling`). That is the real check on the test additions — `bin/run-tests.sh` is Linux-only, so I could not run them locally.
- The first six commits are compile-verified locally on `t-deck-pro`, which builds six of the seven changed source files: `MessageStore`, `MessageRenderer`, `GPSUpdateScheduling`, `MeshPacketQueue`, `RadioLibInterface`, `WebServer`.
- ⚠️ **`ExpressLRSFiveWay.cpp` is not compiled anywhere in this PR's checks.** Its body is behind `INPUTBROKER_EXPRESSLRSFIVEWAY_TYPE`, which only the two `radiomaster_900_bandit*` variants define, and neither is in the PR build matrix. I could not build them locally either (ran the disk out installing the non-S3 ESP32 core libs). The change there is three `Time`/`Throttle` calls in a file that already includes and uses `Throttle`, but it is worth a reviewer's eye or a manual `radiomaster_900_bandit` build before merge.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/mesh/MeshPacketQueue.cpp [hunk 1]
```diff
@@ -186,9 +188,13 @@ bool MeshPacketQueue::replaceLowerPriorityPacket(meshtastic_MeshPacket *p)
...
         // Check if there's a late packet at the queue end
-        auto now = millis();
-        if (backPacket->tx_after < now && (!p->tx_after || backPacket->tx_after > p->tx_after)) {
-            int32_t dt = (int32_t)(backPacket->tx_after - now);
+        const uint32_t now = Time::getMillis();
+        // tx_after is an absolute deadline, so both tests subtract before comparing. A larger
+        // elapsed means the deadline passed longer ago, so the smaller one is scheduled later.
+        const uint32_t backElapsed = now - backPacket->tx_after;
+        const bool newGoesFirst = !p->tx_after || backElapsed < (uint32_t)(now - p->tx_after);
+        if (Throttle::deadlinePassedAt(now, backPacket->tx_after) && newGoesFirst) {
```
[the remark this answers is not in the document]
Comment (thebentern) [reply src/mesh/MeshPacketQueue.cpp]: Good catch — this is a real regression I introduced, not just a style point.

The comparison I replaced (`backPacket->tx_after > p->tx_after`) was correct away from the wrap, and I lost that in the conversion to elapsed times. An elapsed time only orders two deadlines that have **both** passed; a future `p->tx_after` subtracts to a near-2^32 elapsed and reads as the most overdue packet in the queue rather than the least, so a full queue would drop the packet it was about to transmit in favour of one that is not ready.

Fixed in 082a42488 by classifying before ordering, as suggested.

Also added `test/test_meshpacket_queue/` pinning the branch — your `now=1000 / back=900 / p=1100` case is `test_future_incoming_deadline_does_not_evict_an_overdue_packet`, plus both directions of the both-expired ordering, the undelayed arrival, and all of it again with the deadlines and `now` on opposite sides of the wrap.

File: src/input/ExpressLRSFiveWay.cpp [hunk 2]
```diff
@@ -131,8 +131,9 @@ int32_t ExpressLRSFiveWay::runOnce()
     // Do something about this key press
     determineAction((KeyType)keyValue, longPressed ? LONG : SHORT);
 
-    // If there has been recent key activity, poll the joystick slightly more frequently
-    if (now < keyDownStart + (20 * 1000UL)) // Within last 20 seconds
+    // If there has been recent key activity, poll the joystick slightly more frequently. keyDownStart
+    // is 0 until the first press, reading as "recent" for the first 20s of uptime - as it did before.
+    if (Throttle::isWithinTimespanMs(keyDownStart, 20 * 1000UL)) // Within last 20 seconds
```
[the remark this answers is not in the document]
Comment (thebentern) [reply src/input/ExpressLRSFiveWay.cpp]: Taken, in 9a1cdf670.

Worth noting for the record that this is not a regression from this PR — the arithmetic it replaced (`now < keyDownStart + 20000`) classified the zero sentinel the same way, including the once-per-wrap re-trigger. But it is exactly the sentinel-before-the-comparison rule in the coding conventions, and "has there been recent key activity?" has an honest answer when nothing has ever been pressed.

An idle node now starts at the 250ms poll — the documented floor for not missing presses — and moves to 100ms on the first press.

**Comment by NomDeTom** [comment]
The next stage after the lock guard on the time abstraction was to generate standardised methods for the main comparison operators, and insisting on using those throughout the code, so that unsafe comparisons were impossible by design, and so that the intent was implicit in the method chosen.

These would sit alongside the new ones in throttle.cpp, and supplement those, so that you effectively have:
 - Original
   - isWithinTimespanMs
   - deadlinePassed
   - deadlinePassedAt

 - New
   - Deadline::in(ms) 
   - Deadline.armed()
   - Deadline.passed()
   - Deadline.disarm()

I'm also almost certain that another was required, along the lines of Deadline.hasEverPassed() to check if this is the first setting of something, for e.g. GNSS after boot.
